test/src/main/java/org/apache/accumulo/test/zookeeper/ZooKeeperTestingServer.java

```diff
@@ -47,15 +47,15 @@
    * Instantiate a running zookeeper server - this call will block until the server is ready for
    * client connections. It will try three times, with a 5 second pause to connect.
    */
-  public ZooKeeperTestingServer() {
-    this(PortUtils.getRandomFreePort());
+  public ZooKeeperTestingServer(TemporaryFolder tmpDir) {
```

Comment:
This should not accept the TemporaryFolder type, because that is a JUnit specific Rule type. It should instead accept a normal `File` directory, and `TEMP.newFolder()` be called prior to constructing this.

Reply:
> This should not accept the TemporaryFolder type

Why? ZooKeeperTestingServer is a test class and this enforces that a TemporaryFolder is being used as part of the test.  I don't understand the logic here.

Reply:
The JUnit5 conversion was something that was on my radar for this, because I am aware that they got rid of TemporaryFolder. So, I was hoping to limit how much needed to be changed for migrating to JUnit5 regarding this code. If it's not done now, it'll need to be done later, so I figured let's avoid the churn and set ourselves up by using File instead of TemporaryFolder now.

(Also, spotbugs is failing, but I assume you saw that already)

Reply:
Removed TemporaryFolder in f9ed9a8